src/utils/resolve.ts

```diff
@@ -196,14 +290,36 @@ function fullResolve(
         node: settings['import-x/resolve'],
       } // backward compatibility
 
-    for (const { enable, options, resolver } of normalizeConfigResolvers(
+    for (const { enable, name, options, resolver } of normalizeConfigResolvers(
       configResolvers,
       sourceFile,
     )) {
       if (!enable) {
         continue
       }
 
+      // if the resolver is `eslint-import-resolver-node`, we use the new `node` resolver as fallback instead
+      if (LEGACY_NODE_RESOLVERS.includes(name) && !resolver) {
+        const resolverOptions = (options || {}) as NodeResolverOptions
+        const resolved = legacyNodeResolve(
```

Comment:
As we discussed before, it is possible to accidentally result in mixed resolvers being used:

```js
// eslint.config.js
export default [
  {
    settings: {
      'import-x/resolve': optionsWithLegacyFeature
    }
  },
  {
    settings: {
      'import-x/resolve': options
    }
  }
]
```

It is impossible to detect this case, and this will result in unintentional behavior.

Reply:
Without this PR, there is only one built-in default resolver that accepts `import-x/resolve`, `eslint-import-resolver-node`. So when things go wrong, we are definitely sure it is caused by `eslint-import-resolver-node`.

But with this PR, we have two built-in default resolvers, and we wouldn't be sure which resolver is being used and is causing trouble.

Reply:
The search results doesn't mean unsupported options like `packageFilter` are used, I don't know what are you trying to approve.